subsys/net/ip/tcp2.c

```diff
@@ -241,6 +252,12 @@ static int tcp_conn_unref(struct tcp *conn)
 
 	tcp_send_queue_flush(conn);
 
+	if (conn->unacked_data) {
```

Comment:
Silght misunderstanding: perhaps just 'send_data' would be clearer. For unacked_* I meant pending_len.

Reply:
I understood you, but send_data sounded too vague. Wouldn't unacked_data convey the meaning better?

Reply:
I would say, renaming to unacked_data would just add extra confusion.

unacked_data - the data that we already sent, but waiting to be acknowledged. send_data - the overall data that was enqueued to be transmitted. Since we are dealing with peer's mss here, send_data >= unacked_data.

I would say, send_data and unacked_len would be less ambigious here.